demos/http/http_demo_mutual_auth/http_demo_mutual_auth.c

```diff
@@ -136,6 +162,79 @@ static int sendHttpRequest( const TransportInterface_t * pTransportInterface,
 
 /*-----------------------------------------------------------*/
 
+static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext )
+{
+    int returnStatus = EXIT_SUCCESS;
```

Comment:
It is preferable to initialize status variables to failure codes, and then set to success values only when those conditions are met.  Same comment at line 169.  Suggest reworking the logic:

```c
int returnStatus = EXIT_FAILURE;
RetryUtilsStatus_t retryUtilsStatus = RetryUtilsFailure;
.
.
.
do
{
    .
    .
    .
}  while( ( opensslStatus != OPENSSL_SUCCESS ) && ( retryUtilsStatus == RetryUtilsSuccess ) );

if( opensslStatus == OPENSSL_SUCCESS )
{
    returnStatus = EXIT_SUCCESS;
}
else
{
    LogError( ( "Connection to the server failed, all attempts exhausted." ) );
}
return returnStatus;